packages/sit-onyx/src/i18n/index.ts

```diff
@@ -0,0 +1,135 @@
+ */
+const createI18n = (options?: ProvideI18nOptions) => {
+  /**
+   * Current locale.
+   * @default "en-US"
+   */
+  const locale = computed(() => unref(options?.locale) ?? "en-US");
+
+  const messages = computed(() => {
+    if (options?.messages && locale.value in options.messages) {
+      return options.messages[locale.value];
+    }
+    return enUS;
+  });
+
+  /**
+   * Gets the translation for the given key.
+   * If message is not found for current locale, English fallback will be used.
+   * @param placeholders Placeholders that will be replaced in the message string
+   * For pluralization, you must provide the placeholder `n`.
+   */
+  const t = computed(() => {
+    return (
+      key: ObjectToDottedStrings<Translation>,
+      placeholders: Record<string, string | number | undefined> = {},
+    ): string => {
+      // use English message as fallback
+      let message = resolveMessage(key, messages.value) ?? resolveMessage(key, enUS) ?? "";
+
+      if (placeholders) {
+        Object.entries(placeholders).forEach(([key, value]) => {
+          if (value === undefined) return;
+          // "gi" is used to replace all occurrences because String.replaceAll() is not available
+          // in our specified EcmaScript target
+          message = message.replace(new RegExp(`{${key}}`, "gi"), value.toString());
+        });
+      }
+
+      const pluralizationValue = typeof placeholders.n === "number" ? placeholders.n : undefined;
+      return resolvePluralization(message, pluralizationValue);
+    };
+  });
+
+  return { locale, t };
+};
+
+/**
+ * Provides a global i18n instance that is used by Onyx.
+ * Must only be called once in the `App.vue` file.
+ */
+export const provideI18n = (options: ProvideI18nOptions) => {
+  provide(I18N_INJECTION_KEY, createI18n(options));
+};
+
+/**
+ * Injects the Onyx i18n instance.
+ */
+export const injectI18n = () => {
+  const fallbackValue = createI18n();
+  return inject(I18N_INJECTION_KEY, fallbackValue);
+};
+
+/**
+ * Resolves the given dotted key (e.g. `a.b.c`) to the translation value of the given messages.
+ * @returns Message value or undefined if translation does not exist.
+ */
+const resolveMessage = (
+  key: ObjectToDottedStrings<Translation>,
+  messages: DeepPartial<Translation>,
+): string | undefined => {
+  // see https://stackoverflow.com/a/6394168
+  const message = key.split(".").reduce<TranslationValue | undefined>((obj, i) => {
+    if (!obj || typeof obj === "string") return obj;
+    return obj[i];
+  }, messages);
+
+  return message && typeof message === "string" ? message : undefined;
+};
+
+/**
+ * Returns the pluralized message (if multiple formats are defined using `" | "` separator.
+ */
+const resolvePluralization = (message: string, value?: number) => {
+  const formats = message.split(" | ").map((i) => i.trim());
+  if (formats.length <= 1) return message;
+
+  let pluralization: 0 | 1 | 2 = 1;
+  if (value === 0) pluralization = 0;
```

Comment:
```suggestion
  if (!value || value === 0) pluralization = 0;
```

right now an empty value would stay with "1", is this intended?

Reply:
I see the test case, but imho, an empty value is as good as 0 only. we would affect the data integrity by staying with 1 here.

A fancy thing we could do is use pluralization 0 for undefined but writing the "halbgeviertstrich", e.g. in your test:
```
  message = t.value("pluralization" as TestTranslationKey);
  expect(message).toBe("– items");
```
It's UXes trusted friend for these situations 🙃

Reply:
I wanted to align it with `vue-i18n` and there you will get 1 if n is undefined :)

Reply:
did not expect that, but good to know :O better not forget it then ^^